src/ansys/aedt/core/application/aedt_file_management.py

```diff
@@ -133,29 +136,36 @@ def change_objects_visibility(origfile, solid_list):
     newfile = os.path.join(path, "aedttmp.tmp")
 
     if not os.path.isfile(origfile + ".lock"):  # check if the project is closed
-        with open(origfile, "rb") as f, open(newfile, "wb") as n:
-            # Reading file content
-            content = f.read()
-
-            # Searching file content for pattern
-            pattern = re.compile(
-                r"(\$begin 'EditorWindow'\n.+)(Drawings\[.+\])(.+\n\s*\$end 'EditorWindow')", re.UNICODE
-            )
-            # Replacing string
-            # fmt: off
-            view_str = u"Drawings[" + str(len(solid_list)) + u": " + str(solid_list).strip("[")
-            s = pattern.sub(r"\1" + view_str + r"\3", content)
-            # fmt: on
-            # writing file content
-            n.write(str(s))
-
-        # renaming files and deleting temp
-
-        os.remove(origfile)
-        os.rename(newfile, origfile)
-
+        try:
+            # Using text mode with explicit encoding instead of binary mode.
+            with open(origfile, "rb") as f, open(newfile, "wb") as n:
+                # Reading file content
+                content = f.read()
+
+                # Searching file content for pattern
+                pattern = re.compile(
+                    r"(\$begin 'EditorWindow'\n.+)(Drawings\[.+\])(.+\n\s*\$end 'EditorWindow')", re.UNICODE | re.DOTALL
+                )
+                # Replacing string
+                # fmt: off
+                view_str = u"Drawings[" + str(len(solid_list)) + u": " + str(solid_list).strip("[")
+                s = pattern.sub(r"\1" + view_str + r"\3", content)
+                # fmt: on
+                # Writing file content
+                n.write(s)
+            # Renaming files and deleting temporary file
+            os.remove(origfile)
+            os.rename(newfile, origfile)
+            return True
+        except Exception as e:
+            # Cleanup temporary file if exists.
+            if os.path.exists(newfile):
+                os.remove(newfile)
+            logging.error("change_objects_visibility: Error encountered - %s", e)
+            return False
```

Comment:
```suggestion
            raise AEDTRuntimeError("Failed to restrict visibility to specified solids.") from e
```
I think that this will required an additional import